File: starter_templates/sqlite/rust/src/schema.rs
#[derive(Debug)]
pub struct Schema {
    kind: String,
    name: String,
    table_name: String,
    root_page: u8,
    sql: String,
}

impl Schema {
    /// Parses a record into a schema
    pub fn parse(record: Vec<Vec<u8>>) -> Option<Self> {
        let mut items = record.into_iter();
        let kind = items.next()?;
        let name = items.next()?;
        let table_name = items.next()?;
        let root_page = *items.next()?.get(0)?;
        let sql = items.next()?;

        let schema = Self {
            kind: String::from_utf8_lossy(&kind).to_string(),
            name: String::from_utf8_lossy(&name).to_string(),
            table_name: String::from_utf8_lossy(&table_name).to_string(),
            root_page,
            sql: String::from_utf8_lossy(&sql).to_string(),
        };
        Some(schema)
    }
}

File: starter_templates/sqlite/csharp/src/App/Schema.cs
namespace App;

using static System.Text.Encoding;

public record Schema(
    string Type,
    string Name,
    string TableName,
    byte RootPage,
    string Sql)
{
    /// <summary>Parses a record into a schema</summary>
    public static Schema Parse(Record record)
    {
        var type = UTF8.GetString(record.Columns[0].Span);
        var name = UTF8.GetString(record.Columns[1].Span);
        var tableName = UTF8.GetString(record.Columns[2].Span);
        var rootPage = record.Columns[3].Span[0];
        var sql = UTF8.GetString(record.Columns[4].Span);

        return new(type, name, tableName, rootPage, sql);
    }
}
